crates/bindings-csharp/Codegen/Diag.cs

```diff
@@ -34,20 +34,24 @@ internal static class ErrorDescriptor
             field => field
         );
 
-    public static readonly ErrorDescriptor<TypeDeclarationSyntax> EmptyIndexColumns =
+    public static readonly ErrorDescriptor<(
+        ViewIndex index,
+        AttributeData attr
+    )> EmptyIndexColumns =
         new(
             group,
-            "Index attribute must specify index Columns.",
-            table => $"Table {table.Identifier} has an Index.BTree attribute, but no columns.",
-            table => table.BaseList!
+            "Index attribute must specify columns.",
+            ctx =>
+                $"{(ctx.index.AccessorName != "" ? ctx.index.AccessorName : "Index")} has an Index.BTree attribute, but no columns.",
```

Comment:
In database dependency theory, it actually does make sense to key things based on the empty set of columns. It's equivalent to having an index on a column of type `()`. However, it's pretty silly to make such an index, so I'm fine with this error.

We might want to consider whether to allow empty unique constraints. This is semantically meaningful: it means the table can have at most one row. This might be useful if a client wants to have "global state" tables. I opened https://github.com/clockworklabs/SpacetimeDB/issues/2080 to discuss this.